base/operators.jl

```diff
@@ -512,10 +512,10 @@ julia> identity("Well, what did you expect?")
 """
 identity(@nospecialize x) = x
 
-+(x) = x
++(x::Number) = x
 -(x) = Int8(-1)*x
```

Comment:
Kind of seems like all of these other type-unlimited arithmetic fallbacks should also be done away with.